benchmark-service/benchmark/benchmark.py

```diff
@@ -180,30 +183,35 @@ async def compare(request, userdata):  # pylint: disable=unused-argument
         benchmarks_context2 = None
         comparisons = None
     else:
-        benchmarks_context1 = get_benchmarks(file1)
-        benchmarks_context2 = get_benchmarks(file2)
+        benchmarks_context1 = get_benchmarks(app, file1)
+        benchmarks_context2 = get_benchmarks(app, file2)
         comparisons = final_comparisons(get_comparisons(benchmarks_context1, benchmarks_context2, metric))
     context = {'file1': file1,
                'file2': file2,
                'metric': metric,
                'benchmarks1': benchmarks_context1,
                'benchmarks2': benchmarks_context2,
-               'comparisons': comparisons}
+               'comparisons': comparisons,
+               'benchmark_file_list': app['benchmark_file_list']}
     return await render_template('benchmark', request, userdata, 'compare.html', context)
 
 
-def init_app() -> web.Application:
+def on_startup(app):
+    app['gs_reader'] = ReadGoogleStorage(service_account_key_file='/benchmark-gsa-key/key.json')
+    files = app['gs_reader'].list_files()
+    app['benchmark_file_list'] = files
```

Comment:
You don't need the list files here. We talked about this and decided we should just do the query on demand rather than caching the results.